addons/website/static/src/js/editor/snippets.options.js

```diff
@@ -1925,6 +1925,15 @@ options.registry.HeaderNavbar = options.Class.extend({
         if (widgetName === 'option_logo_height_scrolled') {
             return !!this.$('.navbar-brand').length;
         }
+        if (widgetName === 'header_alignment_opt') {
+            let showAlignment = true;
+            if (!this.$target[0].querySelector('.o_offcanvas_menu_toggler')) {
+                // If hamburger type is "Default", hides the alignment option
+                // for "hamburger full" and "magazine" header templates.
+                showAlignment = !this.$target[0].querySelector('#oe_structure_header_hamburger_full_1, #oe_structure_header_magazine_1');
+            }
+            return showAlignment;
```

Comment:
```suggestion
            if (!this.$target[0].querySelector('.o_offcanvas_menu_toggler')) {
                // If hamburger type is "Default", hides the alignment option
                // for "hamburger full" and "magazine" header templates.
                return !this.$target[0].querySelector('#oe_structure_header_hamburger_full_1, #oe_structure_header_magazine_1');
            }
            return true;
```

But strange, is not there many more header where the alignment option does nothing? For the Boxed header for example, what does it do?

Reply:
For Boxed header, the alignment options impact the mobile view.